lib/rules/space-in-parens.js

```diff
@@ -184,9 +188,13 @@ module.exports = function(context) {
 
     return {
         "Program": function checkParenSpaces(node) {
-            var tokens = node.tokens;
-            var prevToken, nextToken;
+            var tokens, prevToken, nextToken;
             exceptions = getExceptions();
+            var comments = sourceCode.getAllComments();
```

Comment:
This is some pretty heavy logic to be doing in a rule. I'd suggest adding this to `SourceCode`. Maybe a tokensAndComments property?